source/common/upstream/upstream_impl.cc

```diff
@@ -1193,6 +1200,14 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
           parent_.updateAllHosts(hosts_added, hosts_removed, locality_lb_endpoint_.priority());
         }
 
+        // TODO(dio): Not sure if we should do this if active health checking is disabled.
```

Comment:
we should, the `all_hosts_` mapping is also used to determine if the new hosts match within the same priority, which affects `hosts_changed`